src/ngraph/ops/batchnorm.h

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef MXNET_NGRAPH_OPS_BATCHNORM_H_
+#define MXNET_NGRAPH_OPS_BATCHNORM_H_
+
+#include <tuple>
+
+#include "ngraph_emitter.h"
+
+namespace ngraph_bridge {
+
+// Create an subgraph of nGraph nodes that's functionally equivalent to nGraph's training-version
+// BatchNorm op, but which doesn't actually use that particular nGraph op.
+// This function exists as a crutch until nGraph's BatchNorm op is fully implemented.
+std::tuple<NgraphNodePtr, NgraphNodePtr, NgraphNodePtr>
+create_batchnorm_training_without_ngraph_bn_op(
+    const float epsilon,
```

Comment:
Can we use a string instead of a float? I don't want to lose precision on multiple data types.

Reply:
Do we expect any values for epsilon that require more precision than `float`?  There's a few reasons I'm doubtful:
- The default value is `0.001f`.
- The extra precision would only be meaningful in graphs that process data more precise than f32 (which I guess just leaves f64).

f32 is all we care about immediately, if our MKL-DNN kernel support in nGraph's BN op is any indication.

And if we need something more precise, I would think we'd want `const double epsilon` here, rather than a string.

Reply:
It's stored as a string in mxnet, so using a string is preserving whatever information is in there already.

Not a huge deal, f32 is probably good enough.